src/main/java/com/deviantart/kafka_connect_s3/AbstractBlockGZIPRecordWriter.java

```diff
@@ -1,110 +1,38 @@
 package com.deviantart.kafka_connect_s3;
 
-import java.io.BufferedWriter;
-import java.io.BufferedWriter;
 import java.io.File;
 import java.io.FileNotFoundException;
 import java.io.FileOutputStream;
 import java.io.FileWriter;
 import java.io.FilterOutputStream;
 import java.io.IOException;
 import java.io.OutputStream;
-import java.io.OutputStreamWriter;
 import java.io.UnsupportedEncodingException;
-
 import java.util.ArrayList;
 import java.util.zip.GZIPOutputStream;
 
 import org.json.simple.JSONArray;
 import org.json.simple.JSONObject;
 
-/**
- * BlockGZIPFileWriter accumulates newline delimited UTF-8 records and writes them to an
- * output file that is readable by GZIP.
- *
- * In fact this file is the concatenation of possibly many separate GZIP files corresponding to smaller chunks
- * of the input. Alongside the output filename.gz file, a file filename-index.json is written containing JSON
- * metadata about the size and location of each block.
- *
- * This allows a reading class to skip to particular line/record without decompressing whole file by looking up
- * the offset of the containing block, seeking to it and beginning GZIp read from there.
- *
- * This is especially useful when the file is an archive in HTTP storage like Amazon S3 where GET request with
- * range headers can allow pulling a small segment from overall compressed file.
- *
- * Note that thanks to GZIP spec, the overall file is perfectly valid and will compress as if it was a single stream
- * with any regular GZIP decoding library or program.
- */
-public class BlockGZIPFileWriter {
-  private String filenameBase;
-  private String path;
-  private GZIPOutputStream gzipStream;
-  private BufferedWriter writer;
-  private CountingOutputStream fileStream;
-
-  private class Chunk {
-    public long rawBytes = 0;
-    public long byteOffset = 0;
-    public long compressedByteLength = 0;
-    public long firstOffset = 0;
-    public long numRecords = 0;
-  };
-
-  private class CountingOutputStream extends FilterOutputStream {
-    private long numBytes = 0;
-
-    CountingOutputStream(OutputStream out) throws IOException {
-      super(out);
-    }
-
-    @Override
-    public void write(int b) throws IOException {
-      out.write(b);
-      numBytes++;
-    }
-    @Override
-    public void write(byte[] b) throws IOException {
-      out.write(b);
-      numBytes += b.length;
-    }
-    @Override
-    public void write(byte[] b, int off, int len) throws IOException {
-      out.write(b, off, len);
-      numBytes += len;
-    }
-
-    public long getNumBytesWritten() {
-      return numBytes;
-    }
-  };
-
-  private ArrayList<Chunk> chunks;
-
+public abstract class AbstractBlockGZIPRecordWriter implements BlockGZIPRecordWriter {
```

Comment:
The core of the change was extracting a base write class so that I could reuse the existing chunking logic while writing raw bytes instead of newline delimited strings.